### PR TITLE
[WIP] Macros 1.1

### PR DESCRIPTION
**Pull request by kvark**

### CODE REVIEW

**Comment by kvark** [comment]
r? anyone

File: src/macros/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#![recursion_limit = "192"]
+
+extern crate proc_macro;
+extern crate syn;
+#[macro_use] extern crate quote;
+
+use proc_macro::TokenStream;
+
+
+#[proc_macro_derive(GfxVertexFormat)]
```
Comment (Bastacyclop) [inline src/macros/src/lib.rs]: is the `Gfx` prefix necessary ? (I hope there is a way to rename a macro on the user side if he has clashes)

File: src/macros/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    let gen = structure(ast, quote!(gfx::shade::Formatted), quote!(gfx::shade::ConstFormat));
+    gen.parse().unwrap()
+}
+
+fn structure(ast: syn::DeriveInput, ty_compile: quote::Tokens, ty_run: quote::Tokens)
+             -> quote::Tokens {
+    let name = &ast.ident;
+    let fields = match ast.body {
+        syn::Body::Struct(syn::VariantData::Struct(ref fields)) => fields,
+        _ => panic!("`GfxStruct` can only be derived for structs"),
```
Comment (Bastacyclop) [inline src/macros/src/lib.rs]: We are not really deriving `GfxStruct`, I think rustc already wraps the derive panics inside a pretty message with the derive name and location.

**Comment by kvark** [comment]
Thanks @Bastacyclop ! Fixed now, merging in an hour.

File: src/macros/src/lib.rs (hunk 2 above)
Comment (kvark) [inline src/macros/src/lib.rs]: Nicely spotted! Fixed in #1207

**Comment by ebkalderon** [comment]
This is fantastic! ❤️ I think this also means that my previous issues with `#![deny(missing_docs)]` is no longer a problem, correct?

**Pull request merged**